railties/lib/rails/generators/rails/master_key/master_key_generator.rb

```diff
@@ -46,7 +44,15 @@ def key_file_generator
         end
 
         def key_ignore
-          [ "", "# Ignore master key for decrypting credentials and more.", "/#{MASTER_KEY_PATH}", "" ].join("\n")
+          [ "", "# Ignore master key for decrypting credentials and more.", "/#{master_key_path}", "" ].join("\n")
+        end
+
+        def master_key_path
+          @master_key_path ||= if Rails.respond_to?(:application)
```

Comment:
Having to check for `Rails.respond_to?(:application)` here is a smell that this is in the wrong place. Generators have configs, so if we want to respect an application config in a generator, the application that should be passing the config to the generate, not the generator asking for the application.